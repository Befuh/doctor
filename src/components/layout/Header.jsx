import React, { Component } from 'react';
import Logo from '../../images/Logo.png';
import { Collapse, Container, Nav, Navbar, NavbarBrand, NavbarToggler, NavItem, NavLink } from 'reactstrap';

import './Header.scss';

class Header extends Component {
  state = {
    isOpen: false
  };

  toggle = event => {
    event.preventDefault();
    this.setState({ isOpen: !this.state.isOpen });
  };

  render () {
    return (
      <Navbar expand='md' color='light' light className='header'>
        <Container>
          <NavbarBrand href='#home'>
            <img src={Logo} alt='Befu'/>
          </NavbarBrand>
          <NavbarToggler onClick={this.toggle}/>
          <Collapse isOpen={this.state.isOpen} navbar>
            <Nav navbar>
              <NavItem>
                <NavLink href='#' className='active navigation-link'>Patients</NavLink>
              </NavItem>
              <NavItem>
                <NavLink href='#' className='navigation-link'>Hospitals</NavLink>
              </NavItem>
            </Nav>
          </Collapse>
        </Container>
      </Navbar>
    );
  }
}

export default Header;
