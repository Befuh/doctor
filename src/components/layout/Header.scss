@import "../../styles/base/variables";
@import "~bootstrap/scss/bootstrap-grid";

.header {
  background-color: $white;
  /**
  * @todo Remove important
   */
  border: none !important;
  padding: 0 !important;

  .header-container {
    border-bottom: solid 1px $light;
  }

  img {
    max-height: 30px;
  }

  .navigation-link{
    font-weight: bold;
    color: $secondary-text-color;
    padding: 1rem 1rem;
    &:hover, &.active {
      color: $primary-text-color;
    }
    &.active{
      border-bottom: solid 2px $primary;
    }
  }

  @include media-breakpoint-up(md){
    .push-right {
      margin-left: 100px;
    }
  }
}
