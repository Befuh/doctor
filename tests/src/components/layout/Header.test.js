import React from 'react';
import { shallow } from 'enzyme';
import Header from '../../../../src/components/layout/Header';

describe('Header', () => {
  let header;
  beforeEach(() => {
    header = shallow(<Header/>)
  });

  it('renders correctly', () => {
    expect(header).toMatchSnapshot();
  });
});
