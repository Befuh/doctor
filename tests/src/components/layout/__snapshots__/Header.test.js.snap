// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header renders correctly 1`] = `
<Navbar
  className="header"
  color="light"
  expand="md"
  light={true}
  tag="nav"
>
  <Container
    tag="div"
  >
    <NavbarBrand
      href="#home"
      tag="a"
    >
      <img
        alt="Befu"
        src="test-file-stub"
      />
    </NavbarBrand>
    <NavbarToggler
      onClick={[Function]}
      tag="button"
      type="button"
    />
    <Collapse
      appear={false}
      enter={true}
      exit={true}
      in={false}
      isOpen={false}
      mountOnEnter={false}
      navbar={true}
      onEnter={[Function]}
      onEntered={[Function]}
      onEntering={[Function]}
      onExit={[Function]}
      onExited={[Function]}
      onExiting={[Function]}
      tag="div"
      timeout={350}
      unmountOnExit={false}
    >
      <Nav
        navbar={true}
        tag="ul"
        vertical={false}
      >
        <NavItem
          tag="li"
        >
          <NavLink
            className="active navigation-link"
            href="#"
            tag="a"
          >
            Patients
          </NavLink>
        </NavItem>
        <NavItem
          tag="li"
        >
          <NavLink
            className="navigation-link"
            href="#"
            tag="a"
          >
            Hospitals
          </NavLink>
        </NavItem>
      </Nav>
    </Collapse>
  </Container>
</Navbar>
`;
